Add ember funcionality

luislopez0690 opened the pull request.

luislopez0690 committed:
first take on ember modals, need to add information for buying

luislopez0690 committed:
almost post working

luislopez0690 committed:
near post, big changes

luislopez0690 committed:
missing liverelo also needs refactoring

missing liverelo also needs refactoring

luislopez0690 committed:
last commit on post method

juandsierra82 commented on app/components/all-books.js:
```diff
@@ -1,4 +1,3 @@
 
-export default Component.extend({
-});
+export default Component.extend({});
```
why do we need this component? comment to let me know why it will be needed or delete if not needed

luislopez0690 replied:
i considered using it in the future, also there is a template for it with some logic to use card arrangement

juandsierra82 commented on app/components/user-table.js:
```diff
@@ -0,0 +1,35 @@
+      let transaction = this.get('store').createRecord('transaction', {
+        user: this.get('model.user'),
+        book: this.set('book', book)
+      })
```
; needed

juandsierra82 commented on app/components/user-table.js:
```diff
@@ -0,0 +1,35 @@
+    addToUserLibrary: function() {
+      let currentTransaction = this.get('transaction')
+      this.sendAction('addToUserLibrary', currentTransaction)
+    },
```
; needed here. i won't comment anymore on this repo but check other places for ; when they are needed

juandsierra82 commented on app/components/user-table.js:
```diff
@@ -0,0 +1,35 @@
+      this.sendAction('addToUserLibrary', currentTransaction)
+    },
+    //currently under development
+    deleteFromUserLibrary: function(book) {
```
that's fine but still show clean code.

juandsierra82 commented on app/models/book.js:
```diff
@@ -1,11 +1,15 @@
\ No newline at end of file
+  name: DS.attr(),
+  author: DS.attr(),
+  sDescription: DS.attr(),
```
we talked about the attribute names... these names don't really tell me much you mean description and summary?

juandsierra82 commented on app/router.js:
```diff
@@ -7,11 +7,11 @@ const Router = EmberRouter.extend({
+  this.route('users', function() {
+    this.route('current', { path: 'current/:user_id' });
+  });
   this.route('books', { path: '/' });
```
any '/' route is the main route so it should be the primary route and all routes should correctly nest under it

juandsierra82 commented on app/routes/transactions.js:
```diff
@@ -1,4 +1,5 @@
 
 export default Route.extend({
-});
+
```
again brief comment that this route is under development or delete if not needed.

juandsierra82 commented on app/routes/users/current.js:
```diff
@@ -0,0 +1,25 @@
+  model(params) {
+    return RSVP.hash({
+      user: this.get('store').findRecord('user', params.user_id),
+      allB: this.get('store').findAll('book')
```
allB? does what? we talked about descriptive yet functional names. also remember this is unnecessary if you nested the routes correctly.

juandsierra82 reviewed (commented):
check my comments i did some additional comments outside the review. please remember that names cannot be "allB" doesn't tell me anything.

luislopez0690 committed:
corrected some name variables

luislopez0690 committed:
final first sprint commit

fangtingliu commented on app/routes/users/current.js:
```diff
@@ -0,0 +1,25 @@
+export default Route.extend({
+  model(params) {
+    return RSVP.hash({
+      user: this.get('store').findRecord('user', params.user_id),
```
be consistent, seems like you are using the following
```
this.store
```

fangtingliu commented on app/templates/components/user-table.hbs:
```diff
@@ -0,0 +1,62 @@
+            {{/each}}
+          </td>
+          <td>
+            {{#each model.user.books as |book|}}
```
why are there two book lists?

fangtingliu commented on app/templates/components/user-table.hbs:
```diff
@@ -0,0 +1,62 @@
+</table>
+<!-- Modal -->
+
+<div class="modal fade" id="exampleModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
```
use a better name for the modal id.

luislopez0690 committed:
improving visual

luislopez0690 committed:
done with sprint 1

luislopez0690 committed:
signing almost working

luislopez0690 committed:
sign up user on the works

fangtingliu commented on app/components/books-deck.js:
```diff
@@ -0,0 +1,29 @@
+  actions: {
+    setModalBook: function(currentBook) {
+      this.set('currentBook', currentBook);
+      let transaction = this.get('store').createRecord('transaction', {
```
let should only be used when your variable will be reassigned later.

luislopez0690 committed:
trying to log in

luislopez0690 committed:
more work on the add user

luislopez0690 committed:
login missing valid

luislopez0690 committed:
login ready

luislopez0690 committed:
merging

Merge branch 'master' of github.com:luislopez0690/bookstore-ember into add-ember-funcionality

juandsierra82 commented on app/components/books-deck.js:
```diff
@@ -0,0 +1,29 @@
+      this.set('transaction', transaction);
+    },
+    addToUserLibrary: function() {
+      const currentTransaction = this.get('transaction')
```
semi colon

juandsierra82 commented on app/components/site-layout.js:
```diff
@@ -1,5 +1,5 @@
 
 export default Component.extend({
-  elementId: 'site-layout'
+  elementId: 'site-layout',
```
get rid of the comma

juandsierra82 commented on app/routes/transactions.js:
```diff
@@ -1,4 +1,6 @@
 import Route from '@ember/routing/route';
-
+///Under Development
```
delete the route

juandsierra82 commented on app/services/request.js:
```diff
@@ -0,0 +1,21 @@
+
+export default Service.extend({
+  request({ method, url, cb, data, errorCb }) {
+    console.log('before sending request data: ', data);
```
dont forget no console logs unless you need em for later

juandsierra82 commented on app/templates/components/all-books.hbs:
```diff
@@ -1,8 +1,12 @@
-<div class="card">
+<!-- Problably will be used later, Under Development-->
+
```
get rid of comment

luislopez0690 committed:
improving UI interaction for the login

fangtingliu commented on app/components/login-form.js:
```diff
@@ -1,4 +1,42 @@
+
+  actions: {
+    login: function(username, password) {
+      let currentUser = this.get('store').createRecord('user', {
```
use const instead

fangtingliu commented on app/components/login-form.js:
```diff
@@ -1,4 +1,42 @@
+      this.sendAction('login', currentUser);
+    },
+    signUp: function(name, username, password) {
+      let newUser = this.get('store').createRecord('user', {
```
same

fangtingliu commented on app/components/login-form.js:
```diff
@@ -1,4 +1,42 @@
+      });
+      this.sendAction('signUp', newUser);
+    },
+    reloadAlert: function() {
```
```
reloadAlert() {
```
optional

fangtingliu commented on app/models/book.js:
```diff
@@ -1,11 +1,15 @@
+  image: DS.attr(),
+  price: DS.attr(),
+  amount: DS.attr(),
+  users: DS.hasMany('user', { async: true }),
```
default is , { async: true }.

optional

fangtingliu commented on app/routes/application.js:
```diff
@@ -1,4 +1,15 @@
 export default Route.extend({
-});
+  model() {
+    let modelHash = {
```
const

fangtingliu commented on app/routes/application.js:
```diff
@@ -1,4 +1,15 @@
-});
+  model() {
+    let modelHash = {
+      loginHref: loginHref,
```
```
obj = {
  loginHref: loginHref
}
```
===
```
obj = {
  loginHref
}
```

luislopez0690 committed:
Fixed PR comments

luislopez0690 committed:
fixed PR comments

luislopez0690 committed:
added custom scss classes
